src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -271,13 +273,27 @@ export const handleGetPublicForm: ControllerHandler<
     case AuthType.SP:
     case AuthType.CP:
       return SpcpFactory.extractJwtPayloadFromRequest(authType, req.cookies)
-        .map(({ userName }) =>
-          res.json({
+        .map(({ userName, exp, iat, rememberMe }) => {
+          if (!exp) {
+            {
+              logger.error({
+                message: 'Invalid expiry time for cookie',
+                meta: logMeta,
+              })
+            }
+            return res.json({ form: publicForm, isIntranetUser })
+          }
+          return res.json({
             form: publicForm,
             isIntranetUser,
-            spcpSession: { userName },
-          }),
-        )
+            spcpSession: {
+              userName,
+              iat,
+              rememberMe,
+              msToExpiry: exp * 1000 - Date.now(), // Used to tell browser when to refresh page upon cookie expiry
```

Comment:
do we need to perform this transformation on the backend? seems to be only relevant to the UI. would it be cleaner if we just returned the relevant contents of the jwt as-is, and the frontend can perform these calculations which are specific to the UI?

Reply:
thanks, edited to return exp directly to frontend